### PR TITLE
Fronts Page skins

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

Adds 'in house' page skin rendering to DCR Fronts

Fixes https://github.com/guardian/dotcom-rendering/issues/5490

## Why?

Parity with Frontend as part of the fronts migration.

Page skins are valuable revenue generators.

### Frontend

For reference this is how we derive `hasPageSkin` server side:

1. In house page skins are configured in Google Ad Manager (GAM)
2. The [PageskinAdAgent](https://github.com/guardian/frontend/blob/9a22d1f6ae2ec439225703502f0318bc04ca6adb/common/app/common/dfp/PageskinAdAgent.scala) job in Frontend polls GAM for page skin line items
3. When creating the [DotcomFrontsRenderingDataModel](https://github.com/guardian/frontend/blob/9a22d1f6ae2ec439225703502f0318bc04ca6adb/common/app/model/dotcomrendering/DotcomFrontsRenderingDataModel.scala) if a page skin's targeting from GAM matches a particular front the property `hasPageSkin` is passed to DCR via the [JavaScriptPage](https://github.com/guardian/frontend/blob/9a22d1f6ae2ec439225703502f0318bc04ca6adb/common/app/views/support/JavaScriptPage.scala#L57) helper and is hence available to DCR server side.

### DCR

This PR takes the following approach if `hasPageSkin: true`:

- constrain `Section`, `FrontSection` and `thrasher` containers to the desktop max-width of 980px
- in lieu of historical browser support for [container queries](https://caniuse.com/css-container-queries) (sigh) `hasPageSkin` is prop drilled to all components that apply media queries locally so they do not apply style or layout changes above a desktop breakpoint
- sets `class="has-page-skin"` on `body` so that ad creative scripts and thrashers know when a page skin is present
- if a page skin is active, for all containers that do not set or override their background colour set a default background colour. So if a page skin sets a background colour it does not bleed into the containers

I decided to constrain containers and _against_ keeping `Section` and `FrontSection` full width and selectively setting background colours in the margins. I think having the containers restricted to the central column keeps the skin and content relatively separated and reduces any layering and interaction concerns. But if there are any better approaches we could use please do comment.

Some thrashers do not currently handle when a page skin is present (e.g wordiply) but @HarryFischer is aware of this issue. It should be a straightforward fix to to use the appropriate [scss mixin](https://github.com/guardian/interactive-atom-thrasher-template/blob/main/shared/css/_mq.scss#L34
).

## Screenshots

| | Before      | After      |
| -- | ----------- | ---------- |
| regular | ![before][] | ![after][] |
| with background colour| ![before][] | ![bgafter][] |

[before]: https://github.com/guardian/dotcom-rendering/assets/7014230/aa6f1d76-34a5-46e9-b615-6f39636235cb
[after]: https://github.com/guardian/dotcom-rendering/assets/7014230/d604f27e-0f99-45aa-b055-902297391626
[bgafter]: https://github.com/guardian/dotcom-rendering/assets/7014230/ecdfb0f7-9957-4bad-a26e-e85713552478

### Video

https://github.com/guardian/dotcom-rendering/assets/7014230/24e2a378-ca3b-4da0-b0fe-e0dc2f7d74f7

### CODE REVIEW

**Review by alinaboghiu — APPROVED** [review]

**Comment by alinaboghiu** [comment]
We reviewed this as a group in a meeting - all looks very neat! 👍

File: dotcom-rendering/src/web/components/AdSlot.tsx [hunk 1]
```diff
@@ -21,14 +22,19 @@ type InlineProps = {
 	index: number;
 	shouldHideReaderRevenue?: boolean;
 	isPaidContent?: boolean;
+	hasPageskin?: boolean;
 };
 
+// TODO move to commercial
```
Comment (domlander) [inline dotcom-rendering/src/web/components/AdSlot.tsx:28]: Was this TODO intended for this PR?

File: dotcom-rendering/src/web/components/Section.tsx [hunk 2]
```diff
@@ -180,6 +183,68 @@ const Content = ({
...
+	hasPageSkin: boolean,
+) => {
+	if (backgroundColour) {
+		return backgroundColour;
+	}
+	if (overrideBackgroundColour) {
+		return overrideBackgroundColour;
+	}
+	if (hasPageSkin) {
+		// TODO check this is the right background colour to use
```
Comment (domlander) [inline dotcom-rendering/src/web/components/Section.tsx:242]: TODO reminder

File: dotcom-rendering/src/web/components/FrontSection.tsx [hunk 3]
```diff
@@ -286,6 +297,20 @@ const titleStyle = css`
...
 
+const decideBackgroundColour = (
+	overrideBackgroundColour: string | undefined,
+	hasPageSkin: boolean,
+) => {
+	if (overrideBackgroundColour) {
+		return overrideBackgroundColour;
+	}
+	if (hasPageSkin) {
+		// TODO check this is the right background colour to use
```
Comment (domlander) [inline dotcom-rendering/src/web/components/FrontSection.tsx:308]: TODO reminder

**Review by domlander — APPROVED** [review]

**Review by Jakeii — APPROVED** [review]

File: dotcom-rendering/src/web/components/AdSlot.tsx (hunk 1 above)
Comment (arelra) [reply dotcom-rendering/src/web/components/AdSlot.tsx]: I will pick this up separately as I didn't want to block this PR for it.

**Pull request merged**